### PR TITLE
Add optional relay mode

### PR DESCRIPTION
**Pull request by tulir**
To enable, set `relay: true` in the config. When enabled, messages from unknown users will be sent to slack prefixed with `**mxid:**`

### CODE REVIEW

File: src/puppetbridge.ts [hunk 1]
```diff
@@ -878,7 +934,13 @@ export class PuppetBridge extends EventEmitter {
 		}
 		const puppetMxid = await this.provisioner.getMxid(room.puppetId);
 		if (event.sender !== puppetMxid) {
-			return; // this isn't our puppeted user, so let's not do anything
+			if (!this.config.relay.enabled) {
+				return; // relaying not enabled, don't allow message from other user
+			} else if (!this.provisioner.canCreate(event.sender)) {
```
Comment (Sorunome) [inline src/puppetbridge.ts]: Shouldn't this maybe be a separate permission? As, if you whitelist many people, many will also be able to create puppet channels then, causing your server to potentially suddenly have tons of slack connections open for all the people maliciously making bridges?

Comment (Sorunome) [reply src/puppetbridge.ts]: Maybe its own blacklist/whitelist for this setting?

File: src/puppetbridge.ts [hunk 2]
```diff
@@ -878,7 +935,13 @@ export class PuppetBridge extends EventEmitter {
 		}
 		const puppetMxid = await this.provisioner.getMxid(room.puppetId);
 		if (event.sender !== puppetMxid) {
-			return; // this isn't our puppeted user, so let's not do anything
+			if (!this.config.relay.enabled) {
```
Comment (Sorunome) [inline src/puppetbridge.ts]: `if (!this.config.relay.enabled || !this.provisioner.canRelay(event.sender)`

File: src/puppetbridge.ts [hunk 3]
```diff
@@ -878,7 +935,13 @@ export class PuppetBridge extends EventEmitter {
...
+				return; // relaying not enabled, don't allow message from other user
+			} else if (!this.provisioner.canRelay(event.sender)) {
+				return; // no permissions to be relayed
+			}
+			await this.applyRelayFormatting(event.room_id, event.sender, event.content);
+			event.sender = puppetMxid;
```
Comment (Sorunome) [inline src/puppetbridge.ts]: what is this needed for?

**Review by Sorunome — COMMENTED** [review]
Looks good, just small nitpicks

File: src/puppetbridge.ts (hunk 3 above)
Comment (tulir) [reply src/puppetbridge.ts]: I wasn't sure if something checks the sender afterwards so I added that just in case

**Pull request merged**